From 3401ce2be280ce4ee77bc1f3c7494501a165e95a Mon Sep 17 00:00:00 2001
From: Victor Zverovich <viz@meta.com>
Date: Tue, 25 Feb 2025 13:32:24 -0800
Subject: [PATCH] Fix ABI compatibility

---
 include/fmt/base.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/fmt/base.h b/include/fmt/base.h
index a37858a8..9ec43c10 100644
--- a/include/fmt/base.h
+++ b/include/fmt/base.h
@@ -2271,7 +2271,7 @@ struct is_output_iterator<
 #endif
 
 // A type-erased reference to an std::locale to avoid a heavy <locale> include.
-struct locale_ref {
+class locale_ref {
 #if FMT_USE_LOCALE
  private:
   const void* locale_;  // A type-erased pointer to std::locale.
@@ -2283,6 +2283,7 @@ struct locale_ref {
   inline explicit operator bool() const noexcept { return locale_ != nullptr; }
 #endif  // FMT_USE_LOCALE
 
+ public:
   template <typename Locale> auto get() const -> Locale;
 };
 
-- 
2.48.1

