From 7ced344c6d5be9de1b2690a3ffe757e7d3ecf7f3 Mon Sep 17 00:00:00 2001
From: Tom de Geus <tom@geus.me>
Date: Fri, 15 Nov 2019 10:30:37 +0100
Subject: [PATCH] Change header to ease Windows build

---
 include/fmt/format.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/fmt/format.h b/include/fmt/format.h
index 9c104df1..5a4668ea 100644
--- a/include/fmt/format.h
+++ b/include/fmt/format.h
@@ -36,7 +36,7 @@
 #include <algorithm>
 #include <cassert>
 #include <cmath>
-#include <cstdint>
+#include <stdint>
 #include <cstring>
 #include <iterator>
 #include <limits>
-- 
2.21.0 (Apple Git-122.2)

