From ad3c7855e2d3ab02ee71205a56334ea1d94f1b81 Mon Sep 17 00:00:00 2001
From: Bruce Mitchener <bruce.mitchener@gmail.com>
Date: Tue, 27 Aug 2019 14:44:55 +0700
Subject: [PATCH 001/186] Fix typos.

---
 ChangeLog.rst        | 2 +-
 include/fmt/chrono.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/ChangeLog.rst b/ChangeLog.rst
index 33a4003f..19e9b552 100644
--- a/ChangeLog.rst
+++ b/ChangeLog.rst
@@ -298,7 +298,7 @@
   `#1222 <https://github.com/fmtlib/fmt/issues/1222>`_).
   Thanks `@alabuzhev (Alex Alabuzhev) <https://github.com/alabuzhev>`_.
 
-* Fixed bugs discovered by fuzzing or during fuzzing integation
+* Fixed bugs discovered by fuzzing or during fuzzing integration
   (`#1124 <https://github.com/fmtlib/fmt/issues/1124>`_,
   `#1127 <https://github.com/fmtlib/fmt/issues/1127>`_,
   `#1132 <https://github.com/fmtlib/fmt/issues/1132>`_,
diff --git a/include/fmt/chrono.h b/include/fmt/chrono.h
index 57ce9ef3..c965cf78 100644
--- a/include/fmt/chrono.h
+++ b/include/fmt/chrono.h
@@ -403,7 +403,7 @@ inline bool isfinite(T value) {
   return std::isfinite(value);
 }
 
-// Convers value to int and checks that it's in the range [0, upper).
+// Converts value to int and checks that it's in the range [0, upper).
 template <typename T, FMT_ENABLE_IF(std::is_integral<T>::value)>
 inline int to_nonnegative_int(T value, int upper) {
   FMT_ASSERT(value >= 0 && value <= upper, "invalid value");
-- 
2.21.0 (Apple Git-122.2)

