From 168460f02c0f720ea4a1bfef82116632ce7f2d9e Mon Sep 17 00:00:00 2001
From: Victor Zverovich <victor.zverovich@gmail.com>
Date: Tue, 3 Dec 2019 06:45:00 -0800
Subject: [PATCH 6/6] Remove TYPES

---
 include/fmt/core.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/include/fmt/core.h b/include/fmt/core.h
index 80e87e21..d37486d0 100644
--- a/include/fmt/core.h
+++ b/include/fmt/core.h
@@ -1211,7 +1211,6 @@ template <typename Context, typename... Args> class format_arg_store {
   static constexpr unsigned long long types =
       is_packed ? internal::encode_types<Context, Args...>()
                 : internal::is_unpacked_bit | num_args;
-  FMT_DEPRECATED static constexpr unsigned long long TYPES = types;
 
   format_arg_store(const Args&... args)
       : data_{internal::make_arg<is_packed, Context>(args)...} {}
-- 
2.21.0 (Apple Git-122.2)

