From 123e7f7fc33381a07a9d4d28ea2006f5132c503d Mon Sep 17 00:00:00 2001
From: Victor Zverovich <victor.zverovich@gmail.com>
Date: Tue, 3 Dec 2019 09:24:15 -0800
Subject: [PATCH 7/7] Revert #1433 because of build failures (#1450)

---
 include/fmt/core.h | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/include/fmt/core.h b/include/fmt/core.h
index d37486d0..e766b3a9 100644
--- a/include/fmt/core.h
+++ b/include/fmt/core.h
@@ -166,15 +166,10 @@
 #endif
 
 #if !defined(FMT_HEADER_ONLY) && defined(_WIN32)
-#  if FMT_MSC_VER
-#    define FMT_NO_W4275 __pragma(warning(suppress : 4275))
-#  else
-#    define FMT_NO_W4275
-#  endif
 #  ifdef FMT_EXPORT
-#    define FMT_API FMT_NO_W4275 __declspec(dllexport)
+#    define FMT_API __declspec(dllexport)
 #  elif defined(FMT_SHARED)
-#    define FMT_API FMT_NO_W4275 __declspec(dllimport)
+#    define FMT_API __declspec(dllimport)
 #    define FMT_EXTERN_TEMPLATE_API FMT_API
 #  endif
 #endif
-- 
2.21.0 (Apple Git-122.2)

