From 2730e901865a401c75e4881a27350108a6d4f0ed Mon Sep 17 00:00:00 2001
From: Victor Zverovich <victor.zverovich@gmail.com>
Date: Wed, 9 Oct 2019 18:58:40 -0700
Subject: [PATCH 067/186] Fix compile error in printf with gcc9 (#1354)

---
 include/fmt/printf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/fmt/printf.h b/include/fmt/printf.h
index 7d49af45..8be33b99 100644
--- a/include/fmt/printf.h
+++ b/include/fmt/printf.h
@@ -235,7 +235,7 @@ class printf_arg_formatter : public internal::arg_formatter_base<Range> {
   printf_arg_formatter(iterator iter, format_specs& specs, context_type& ctx)
       : base(Range(iter), &specs, internal::locale_ref()), context_(ctx) {}
 
-  template <typename T, FMT_ENABLE_IF(std::is_integral<T>::value)>
+  template <typename T, FMT_ENABLE_IF(fmt::internal::is_integral<T>::value)>
   iterator operator()(T value) {
     // MSVC2013 fails to compile separate overloads for bool and char_type so
     // use std::is_same instead.
-- 
2.21.0 (Apple Git-122.2)

