From d6eede9e085f0b36edcf0a2f6dff5f7875181019 Mon Sep 17 00:00:00 2001
From: Victor Zverovich <victor.zverovich@gmail.com>
Date: Fri, 1 Nov 2019 12:09:04 -0700
Subject: [PATCH 106/186] Remove redundant ctor

---
 include/fmt/format.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/fmt/format.h b/include/fmt/format.h
index 4c8e32b7..6ed7fc43 100644
--- a/include/fmt/format.h
+++ b/include/fmt/format.h
@@ -2178,8 +2178,7 @@ template <typename Char> struct arg_ref {
 
   arg_id_kind kind;
   union value {
-    FMT_CONSTEXPR value() : index{0u} {}
-    FMT_CONSTEXPR value(int id) : index{id} {}
+    FMT_CONSTEXPR value(int id = 0) : index{id} {}
     FMT_CONSTEXPR value(basic_string_view<Char> n) : name(n) {}
 
     int index;
-- 
2.21.0 (Apple Git-122.2)

