From e9bff7881434d1ed42b07cf352d0cedb94851775 Mon Sep 17 00:00:00 2001
From: Victor Zverovich <victor.zverovich@gmail.com>
Date: Tue, 19 Nov 2019 11:54:18 -0800
Subject: [PATCH 145/186] Don't parse % unless FMT_DEPRECATED_PERCENT is set

---
 include/fmt/format.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/fmt/format.h b/include/fmt/format.h
index efdb211b..4126ab4b 100644
--- a/include/fmt/format.h
+++ b/include/fmt/format.h
@@ -1288,12 +1288,12 @@ FMT_CONSTEXPR float_spec parse_float_type_spec(char spec,
   case 'f':
     result.format = float_format::fixed;
     break;
-  case '%':
 #if FMT_DEPRECATED_PERCENT
+  case '%':
     result.format = float_format::fixed;
     result.percent = true;
-#endif
     break;
+#endif
   case 'A':
     result.upper = true;
     FMT_FALLTHROUGH;
-- 
2.21.0 (Apple Git-122.2)

